frontend/src/Editor/LeftSidebar/SidebarPageSelector/index.jsx

```diff
@@ -96,7 +103,10 @@ const LeftSidebarPageSelector = ({
               </Button>
               <Button
                 title={`${pinned ? 'Unpin' : 'Pin'}`}
-                onClick={() => setPinned(!pinned)}
+                onClick={() => {
+                  setPinned(!pinned);
+                  setHaveUserPinned(true);
```

Comment:
```suggestion
                  !haveUserPinned && setHaveUserPinned(true);
```

Have a check and update the state if the value is different